torch_geometric/transforms/rooted_subgraphs.py

```diff
@@ -0,0 +1,210 @@
+import torch
+from torch_cluster import random_walk
+from torch_sparse import SparseTensor
+
+from torch_geometric.data import Data
+from torch_geometric.transforms import BaseTransform
+
+
+class RootedSubgraphsData(Data):
```

Comment:
I'm wondering if it would make more sense to implement this as a loader rather than a new `Data` type. It'd be a fairly big change to the current structure of this feature though, and I didn't yet understand all the details.

What do you think @rusty1s ?

Reply:
Not sure I fully understand what you mean. Can you explain more? :)

Reply:
My initial thought was that the `RootedSubgraphsData` is storing a lot of duplicate information, and it might be better to compute the subgraph while iterating through the dataset in a loader, rather than upfront transforming `data`.  

Does that make any sense? Or did I misunderstand how the `Data` works with transforms?

Reply:
The transform can either be applied beforehand or during data loading, so I guess your intuition is already somewhat supported by implementing this as a transform?

Reply:
If its applied during data loading, it will still create all of the rooted subgraphs at the beginning of the interation right?

Reply:
Sorry I realized I misunderstand while I was trying to write tests - I had it in my mind we wanted to formulate a minibatch over different rooted subgraphs. Its clear now, lets leave it as is.

Reply:
Yeah, we need all in order to enhance expressive power, right?